pkg/gardenlet/operation/shoot/shoot.go

```diff
@@ -164,6 +164,12 @@ func (b *Builder) WithShootCredentialsFrom(c client.Reader) *Builder {
 	return b
 }
 
+// WithoutShootCredentials sets the shootCredentialsFunc attribute at the builder to return `nil` as credentials.
+func (b *Builder) WithoutShootCredentials() *Builder {
+	b.shootCredentialsFunc = func(context.Context, string, string, bool) (client.Object, error) { return nil, nil }
```

Comment:
If I'm not mistaken, this could lead to an error when building the operation [here](https://github.com/gardener/gardener/blob/d4250c11cb5eabf85f48fce830174345de2543f0/pkg/gardenlet/operation/shoot/shoot.go#L248). Can you double check please?

Reply:
The shoot controller is setting the credentials and it should be working fine:
https://github.com/gardener/gardener/blob/d4250c11cb5eabf85f48fce830174345de2543f0/pkg/gardenlet/controller/shoot/shoot/reconciler.go#L363

The shoot care controller on the other side is not using the credentials anywhere as of now, thus it is changed to not try to read them from the garden cluster.

Reply:
No matter if the care controller needs the credentials or not, please check the following call stack:

https://github.com/gardener/gardener/blob/ab89d09e2d6b2cb6d34b46bcc2186a94020b6a9e/pkg/gardenlet/controller/shoot/care/reconciler.go#L109

The operation invokes `WithShootFromCluster`

https://github.com/gardener/gardener/blob/ab89d09e2d6b2cb6d34b46bcc2186a94020b6a9e/pkg/gardenlet/operation/operation.go#L146

`WithShootFromCluster` tries to build the `shoot` object:

https://github.com/gardener/gardener/blob/ab89d09e2d6b2cb6d34b46bcc2186a94020b6a9e/pkg/gardenlet/operation/operation.go#L160

The `Build` function tries to process the credentials:

https://github.com/gardener/gardener/blob/ab89d09e2d6b2cb6d34b46bcc2186a94020b6a9e/pkg/gardenlet/operation/shoot/shoot.go#L248

This will error for some shoots:

https://github.com/gardener/gardener/blob/ab89d09e2d6b2cb6d34b46bcc2186a94020b6a9e/pkg/utils/gardener/shoot.go#L616